e2e/launcher/pocketserver.go

```diff
@@ -0,0 +1,124 @@
+package launcher
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+)
+
+type PocketServer interface {
+	Start(...string) error
+	Kill() error
+	RegisterPatternActor(patternActor PatternActor, stream Stream) error
+}
+
+func NewPocketServer(executableLocation string) PocketServer {
+	return &pocketServer{
+		executableLocation: executableLocation,
+		pocketInstance:     nil,
+	}
+}
+
+type pocketServer struct {
+	executableLocation string
```

Comment:
By location, do we mean `Path`? If so, could/should we rename it?

Reply:
yes to all

Reply:
Please update